File: Java/EstruturaDeRepeticao/Exercicio35.java
// Encontrar números primos é uma tarefa difícil.
// Faça um programa que gera uma lista dos números primos existentes entre 1 e um número inteiro informado pelo usuário.
package EstruturaDeRepeticao;
import java.util.Scanner;

public class Exercicio35 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Digite um número inteiro: ");
        int numero = scanner.nextInt();

        System.out.println("Números primos entre 1 e " + numero + ":");

        for (int i = 2; i <= numero; i++) {
            if (primeNumber(i)) {
                System.out.print(i + " ");
            }
        }

        scanner.close();
    }

    public static boolean primeNumber(int numero) {
        if (numero <= 1) {
            return false;
        }

        for (int i = 2; i <= Math.sqrt(numero); i++) {
            if (numero % i == 0) {
                return false;
            }
        }

        return true;
    }
}

File: Python/EstruturaDeRepeticao/exercicio_35.py
"""
Encontrar números primos é uma tarefa difícil.
Faça um programa que gera uma lista dos números primos existentes entre 1 e um número inteiro informado pelo usuário.
"""
def prime_number(numero):
    if numero < 2:
        return False

    for i in range(2, int(numero ** 0.5) + 1):
        if numero % i == 0:
            return False

    return True

numero_limite = int(input("Digite um número inteiro limite: "))

primos = [numero for numero in range(2, numero_limite + 1) if prime_number(numero)]

print(f"Números primos entre 1 e {numero_limite}: {primos}")
